plugin-niassa+pinery/src/main/java/ca/on/oicr/gsi/shesmu/niassa/Bcl2FastqSampleEntry.java

```diff
@@ -0,0 +1,81 @@
+package ca.on.oicr.gsi.shesmu.niassa;
+
+import ca.on.oicr.gsi.provenance.model.LimsKey;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.function.ToIntFunction;
+import java.util.regex.Pattern;
+
+public class Bcl2FastqSampleEntry implements LimsKey {
+  private final String barcode;
+  private final String groupId;
+  private final String limsId;
+  private final String limsProvider;
+  private final ZonedDateTime limsTimestamp;
+  private final String limsVersion;
+  private final String sampleId;
+
+  public Bcl2FastqSampleEntry(String barcode, String sampleId, LimsKey limsKey, String groupId) {
```

Comment:
Change sampleId to libraryName for consistency?

Reply:
Will do.